patches/server/0726-Added-PlayerAngerPiglinByInteractionEvent.patch

```diff
@@ -0,0 +1,133 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Mon, 4 Jan 2021 23:47:30 -0800
+Subject: [PATCH] Added PlayerAngerPiglinByInteractionEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/monster/piglin/PiglinAi.java b/src/main/java/net/minecraft/world/entity/monster/piglin/PiglinAi.java
+index 01d307e0022656509bc8fd02d6bac78a6af3a7be..d74e2386d87e677827351dbc021becd095297235 100644
+--- a/src/main/java/net/minecraft/world/entity/monster/piglin/PiglinAi.java
++++ b/src/main/java/net/minecraft/world/entity/monster/piglin/PiglinAi.java
+@@ -468,11 +468,28 @@ public class PiglinAi {
+     }
+ 
+     public static void angerNearbyPiglins(Player player, boolean blockOpen) {
++        // Paper start
++        PiglinAi.angerNearbyPiglins(player, blockOpen, null, null);
++    }
++    public static void angerNearbyPiglins(Player player, boolean blockOpen, @javax.annotation.Nullable org.bukkit.Location location, @javax.annotation.Nullable org.bukkit.inventory.InventoryHolder inventoryHolder) {
++        // Paper end
+         List<AbstractPiglin> list = (List) player.level.getEntitiesOfClass(Piglin.class, player.getBoundingBox().inflate(16.0D)); // CraftBukkit - decompile error
+ 
+-        list.stream().filter(PiglinAi::isIdle).filter((entitypiglin) -> {
+-            return !blockOpen || BehaviorUtils.canSee((LivingEntity) entitypiglin, player);
+-        }).forEach((entitypiglin) -> {
++        // Paper start - replace streams and call event
++        List<org.bukkit.entity.Piglin> piglinsToBeAngered = com.google.common.collect.Lists.newArrayList();
++        for (AbstractPiglin entitypiglin : list) {
++            if (PiglinAi.isIdle(entitypiglin) && (!blockOpen || BehaviorUtils.canSee(entitypiglin, player))) {
++                piglinsToBeAngered.add((org.bukkit.entity.Piglin) entitypiglin.getBukkitEntity()); // safe cast because the original list gets the Piglin class
++            }
++        }
++        if (location != null) {
++            io.papermc.paper.event.player.PlayerAngerPiglinByInteractionEvent event = new io.papermc.paper.event.player.PlayerAngerPiglinByInteractionEvent((org.bukkit.entity.Player) player.getBukkitEntity(), location, blockOpen, inventoryHolder, piglinsToBeAngered);
++            if (!event.callEvent()) return;
++            piglinsToBeAngered = event.getPiglinsToBeAngered();
```

Comment:
We allow modifying the contents of the list in the event, but I don't see a setter for the list itself. This shouldn't be necessary?

Reply:
correct